src/main/java/gyro/google/compute/ComputeAutoscalingPolicy.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.services.compute.model.AutoscalingPolicy;
+import com.google.api.services.compute.model.AutoscalingPolicyCustomMetricUtilization;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Min;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+
+public class ComputeAutoscalingPolicy extends Diffable implements Copyable<AutoscalingPolicy> {
+
+    private Integer coolDownPeriodSec;
+
+    private ComputeAutoscalingPolicyCpuUtilization cpuUtilization;
+
+    private List<ComputeAutoscalingPolicyCustomMetricUtilization> customMetricUtilization;
+
+    private ComputeAutoscalingPolicyLoadBalancingUtilization loadBalancingUtilization;
+
+    private Integer maxNumReplicas;
+
+    private Integer minNumReplicas;
+
+    private String mode;
+
+    /**
+     * The number of seconds that the autoscaler should wait before it starts collecting information from a new instance.
+     * This prevents the autoscaler from collecting information when the instance is initializing, during which the collected usage would not be reliable. The default time autoscaler waits is 60 seconds. Virtual machine initialization times might vary because of numerous factors. We recommend that you test how long an instance may take to initialize. To do this, create an instance and time the startup process.
```

Comment:
```suggestion
     * This prevents the autoscaler from collecting information when the instance is initializing, during which the collected usage would not be reliable. The default time autoscaler waits is ``60`` seconds. Virtual machine initialization times might vary because of numerous factors. We recommend that you test how long an instance may take to initialize. To do this, create an instance and time the startup process.
```